src/result_processor.c

```diff
@@ -954,7 +954,46 @@ static ResultProcessor *RPSafeLoader_New(RedisSearchCtx *sctx, RLookup *lk, cons
 
 /*********************************************************************************/
 
+typedef struct {
+  ResultProcessor base;
+  const RLookupKey *out;
+} RPKeyNameLoader;
+
+static inline void RPKeyNameLoader_Free(ResultProcessor *self) {
+  rm_free(self);
+}
+
+static int RPKeyNameLoader_Next(ResultProcessor *base, SearchResult *res) {
+  int rc = base->upstream->Next(base->upstream, res);
+  if (RS_RESULT_OK == rc) {
+    RPKeyNameLoader *nl = (RPKeyNameLoader *)base;
+    size_t keyLen = sdslen(res->dmd->keyPtr); // keyPtr is an sds
+    RLookup_WriteOwnKey(nl->out, &res->rowdata, RS_NewCopiedString(res->dmd->keyPtr, keyLen));
+  }
+  return rc;
+}
+
+static ResultProcessor *RPKeyNameLoader_New(const RLookupKey *key) {
+  RPKeyNameLoader *rp = rm_calloc(1, sizeof(*rp));
+  rp->out = key;
+
+  ResultProcessor *base = &rp->base;
+  base->Free = RPKeyNameLoader_Free;
+  base->Next = RPKeyNameLoader_Next;
+  base->type = RP_NAME_LOADER;
+  return base;
+}
+
+/*********************************************************************************/
+
 ResultProcessor *RPLoader_New(AREQ *r, RLookup *lk, const RLookupKey **keys, size_t nkeys, bool forceLoad) {
+  if (RSGlobalConfig.enableUnstableFeatures) {
+    if (nkeys == 1 && !strcmp(keys[0]->path, UNDERSCORE_KEY)) {
+      // Return a thin RP that doesn't actually loads anything or access to the key space
+      // Returning without turning on the `QEXEC_S_HAS_LOAD` flag
+      return RPKeyNameLoader_New(keys[0]);
```

Comment:
How is it reported as not covered?

Reply:
I think something got messed up when I rebased the branch from 2.10 to master. The test runs and passes on the coverage test so we surly use this code